databricks/koalas/series.py

```diff
@@ -1452,6 +1452,109 @@ def clip(self, lower: Union[float, int] = None, upper: Union[float, int] = None)
+
+        Remove elements of a Series based on specifying the index labels.
+        When using a multi-index, labels on different levels can be removed by specifying the level.
+
+        Parameters
+        ----------
+        labels : single label or list-like
+            Index labels to drop.
+        axis : 0, default 0
+            Redundant for application on Series.
+        index : None
+            Redundant for application on Series, but index can be used instead of labels.
+        level : int or level name, optional
+            For MultiIndex, level for which the labels will be removed.
+
+        Returns
+        -------
+        Series
+            Series with specified index labels removed.
+
+        See Also
+        --------
+        Series.dropna
+
+        Examples
+        --------
+        >>> s = ks.Series(data=np.arange(3), index=['A', 'B', 'C'])
+        >>> s
+        A    0
+        B    1
+        C    2
+        Name: 0, dtype: int64
+
+        Drop single label A
+
+        >>> s.drop('A')
+        B    1
+        C    2
+        Name: 0, dtype: int64
+
+        Drop labels B and C
+
+        >>> s.drop(labels=['B', 'C'])
+        A    0
+        Name: 0, dtype: int64
+
+        Also support for MultiIndex
+
+        >>> midx = pd.MultiIndex(levels=[['lama', 'cow', 'falcon'],
+        ...                              ['speed', 'weight', 'length']],
+        ...                      codes=[[0, 0, 0, 1, 1, 1, 2, 2, 2],
+        ...                             [0, 1, 2, 0, 1, 2, 0, 1, 2]])
+        >>> s = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3],
+        ...               index=midx)
+        >>> s
+        lama    speed      45.0
+                weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(labels='weight', level=1)
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        Notes
+        -----
+        Currently only axis = 0 is supported in this function,
+        axis = 1 is yet to be implemented.
+        """
+        if labels is not None:
+            axis = DataFrame._validate_axis(axis)
+            if axis == 0:
+                return self.drop(index=labels, level=level)
+            raise NotImplementedError("Drop currently only works for axis=0")
+        elif index is not None:
+            if isinstance(index, str):
+                index = [index]
+
+            if level is None:
+                level = 0
+
+            sdf = self._internal.sdf.where(
+                ~F.col('__index_level_{}__'.format(level)).isin(index))
```

Comment:
`F.col('__index_level_{}__'.format(level))` -> `self._internal.index_scols[level]`?
The index column name is not always the form of `'__index_level_{}__'`.

Btw, we need to check the range of `level` to avoid "out of bounds".

Reply:
@ueshin Fantastic. i didn't know it. Thanks for comment!! 😃